src/Operation/CreateCollection.php

```diff
@@ -170,6 +174,10 @@ public function __construct($databaseName, $collectionName, array $options = [])
             unset($options['writeConcern']);
         }
 
+        if (isset($options['autoIndexId'])) {
+            trigger_error('The "autoIndexId" option is deprecated and will be removed in a future release', E_USER_DEPRECATED);
```

Comment:
Do you want to show deprecations by default? For reference, Symfony silences these `trigger_error` calls to make deprecation warnings opt-in instead of opt-out: without silencing, the default PHP error handler will start printing notices about deprecations, even if the user has no intention of fixing them or listening to them. Using the silencing operator would effectively require the user to use some kind of deprecation listener (such as the one provided by Symfony) to "listen" to these deprecation warnings.

Reply:
> I assume it stops PHP's default handler from emitting noise but doesn't actually prevent a custom error handler from being notified?

Your assumption is correct. Symfony changed this behavior shortly after introducing deprecation warnings in Symfony 2.7 after complaints about large number of deprecation notices being found in logs. The discussion around the change can be found in https://github.com/symfony/symfony/pull/14900.

Reply:
Created [PHPLIB-347](https://jira.mongodb.org/browse/PHPLIB-347) to track this further. Thanks for the suggestion!